app/src/main/java/com/kickstarter/services/interceptors/GraphQLInterceptor.kt

```diff
@@ -5,20 +5,25 @@ import com.kickstarter.libs.Build
 import com.kickstarter.libs.CurrentUserType
 import com.kickstarter.libs.utils.WebUtils
 import okhttp3.Interceptor
+import okhttp3.Interceptor.Chain
 import okhttp3.Response
 
 class GraphQLInterceptor(private val clientId: String,
                          private val currentUser: CurrentUserType,
                          private val build: Build) : Interceptor {
-    override fun intercept(chain: Interceptor.Chain?): Response {
-        val original = chain!!.request()
-        val builder = original.newBuilder().method(original.method(), original.body())
-        if (this.currentUser.exists()) {
-            builder.addHeader("Authorization", "token " + this.currentUser.accessToken)
-        }
+    override fun intercept(chain: Chain): Response {
+        val original = chain.request()
+        val builder = original.newBuilder().method(original.method, original.body)
+
+        this.currentUser.observable()
+                .subscribe {
+                    builder.addHeader("Authorization", "token " + this.currentUser.accessToken)
```

Comment:
What is exactly happening here? I understand that when this `currentUser.observable()` emits, we're adding a header to the chain builder, but what if this emits after the this method is executed and returns the chain, what happens?

Reply:
this.currentUser.exists() is deprecated, so changing it over .observable